onnxruntime/core/providers/qnn/builder/opbuilder/slice_op_builder.cc

```diff
@@ -75,6 +72,45 @@ void SliceOpBuilder::GetDataFromAttribute(const NodeUnit& node_unit,
   }
 }
 
+// Gets the data from initializer inputs (e.g., starts, ends, axes, or steps) as a TensorShapeVector.
+static Status GetInitializerInputData(const NodeUnitIODef& input, const QnnModelWrapper& qnn_model_wrapper,
+                                      TensorShapeVector& output) {
+  OnnxInputInfo input_info = {};
+  ORT_RETURN_IF_ERROR(qnn_model_wrapper.GetOnnxInputInfo(input, input_info));
+  ORT_RETURN_IF_NOT(input_info.is_initializer,
+                    "QNN requires the starts, ends, axes, and steps inputs to "
+                    "be initializers");
+
+  std::vector<uint8_t> initializer_bytes;
+  ORT_RETURN_IF_ERROR(qnn_model_wrapper.UnpackInitializerData(*input_info.initializer_tensor, initializer_bytes));
+
+  size_t tensor_byte_size = initializer_bytes.size();
+  const auto data_type = input_info.initializer_tensor->data_type();
+
+  Status status;
+
+  switch (data_type) {
+    case ONNX_NAMESPACE::TensorProto_DataType_INT64: {
+      const int64_t* tensor_data = reinterpret_cast<const int64_t*>(initializer_bytes.data());
+      size_t size = tensor_byte_size / sizeof(int64_t);
```

Comment:
Are we expecting to fetch a fixed amount of information? If so, perhaps we could do some checks?

Reply:
Yes, we are expecting a fixed amount of information (based on number of bytes and size of each element). I created a helper function that reinterprets bytes to a span of larger elements in a way that is less error-prone.